tests/test_pulse.py

```diff
@@ -270,3 +270,50 @@ def test_field_envelope_shifted(make_pulse, envelope_type):
     assert envelope[0] == pytest.approx(envelope[-1], abs=1e-15)
     assert envelope[1] == pytest.approx(envelope[-2], abs=1e-15)
     assert envelope[2] == pytest.approx(envelope[-3], abs=1e-15)
+
+
+@pytest.mark.parametrize("envelope_type", ENVELOPE_TYPES)
+def test_pulse_wigner(make_pulse, envelope_type):
+    pulse = make_pulse(envelope_type=envelope_type)
+    tprime = 1.0
+    de = 0.2
+
+    integral = pulse.pulse_wigner(tprime, de)
+
+    s = snapshot(
+        {
+            "gauss": 0.06628966759244047,
+            "lorentz": 0.44606065146635454,
+            "sech": 0.2328991679544244,
+            "sin": 1.4997597826618577e-35,
```

Comment:
Somewhat surprising that the value for `sin` pulse is so much different.

Should we test for a larger set of `de` and `tprime` values?